### ISSUE TITLE
Rollback may not happen

### ISSUE THREAD

**Issue by arnaud-lb** [meadsteve/Tale#9]
Rollback will not happen if the PHP process crashes during a transaction, which will eventually happen. A PHP process can crash in case of fatal error (memory limit and others), outside signal, OS crash, VM shutdown, power outage, etc.

The rollback steps themselves may also fail if one of the services is temporarily down.

I believe that orchestrators should handle this, otherwise the pattern would be efficiently replaced with a try/catch.

**Comment by meadsteve** [comment]
@arnaud-lb this is correct. As it's currently implemented this was a way for me to have compostable try-catch like behaviour without complicated nesting.

In order to handle the case you've described we'd need some external store of state. I'd approve a PR with this but it's not something I'm currently planning on implementing myself.
